### PR TITLE
USHIFT-363: Use route CRD from cluster-ingress-operator repo in rebase script

### PR DESCRIPTION
**Pull request by oglok**
Signed-off-by: Ricardo Noriega <rnoriega@redhat.com>

**Which issue(s) this PR addresses**:

Closes [USHIFT-363](https://issues.redhat.com//browse/USHIFT-363)

### CODE REVIEW

File: assets/crd/route.crd.yaml [hunk 1]
```diff
@@ -69,10 +69,6 @@ spec:
               - properties:
                   path:
                     maxLength: 0
-              - properties:
```
Comment (dhellmann) [inline assets/crd/route.crd.yaml:72]: Weren't these lines necessary to fix a bug in the past?

**Comment by fzdarsky** [comment]
The `openshift/api` version differs between components and therefore potentially also the route CRD, e.g. currently

    cluster-ingress-operator/go.mod:	github.com/openshift/api v0.0.0-20220801162917-f4ac46b878fe
    kubernetes/go.mod:	github.com/openshift/api v0.0.0-20220525145417-ee5b62754c68

Are we certain the one from o/k gets applied and not the one from CIO?

File: assets/crd/route.crd.yaml (hunk 1 above)
Comment (oglok) [reply assets/crd/route.crd.yaml:72]: I see, you're talking about this one: https://github.com/openshift/microshift/commit/7506328bde4effc43620c80772b22548305165a8

I'll fix it.

Comment (oglok) [reply assets/crd/route.crd.yaml:72]: > The `openshift/api` version differs between components and therefore potentially also the route CRD, e.g. currently
> 
> ```
> cluster-ingress-operator/go.mod:	github.com/openshift/api v0.0.0-20220801162917-f4ac46b878fe
> kubernetes/go.mod:	github.com/openshift/api v0.0.0-20220525145417-ee5b62754c68
> ```
> 
> Are we certain the one from o/k gets applied and not the one from CIO?

The canonical location of this CRD is https://github.com/openshift/api/blob/master/route/v1/route.crd.yaml and according to the Network Edge team, this is the only one maintained. They don't control the files vendored by operators (except CIO). So my suggestion would be to clone the api repo as part of the rebase procedure. Does it make sense ?

Comment (dhellmann) [reply assets/crd/route.crd.yaml:72]: Yeah, we should use the version of the file they consider canonical.

**Comment by oglok** [comment]
> The `openshift/api` version differs between components and therefore potentially also the route CRD, e.g. currently
> 
> ```
> cluster-ingress-operator/go.mod:	github.com/openshift/api v0.0.0-20220801162917-f4ac46b878fe
> kubernetes/go.mod:	github.com/openshift/api v0.0.0-20220525145417-ee5b62754c68
> ```
> 
> Are we certain the one from o/k gets applied and not the one from CIO?

It'd be better to use the original location of the route CRD, however, due to the approach of the rebase script (pulling from container images from the release payload) we can use the vendored file from the cluster-ingress-operator.

File: pkg/assets/bindata.go [hunk 2]
```diff
@@ -0,0 +1,6928 @@
+// Code generated for package assets by go-bindata DO NOT EDIT. (@generated)
```
Comment (pmtk) [inline pkg/assets/bindata.go:1]: Please remove this file

Comment (pmtk) [inline pkg/assets/bindata.go:1]: Please remove this file

**Review by pmtk — CHANGES_REQUESTED** [review]

File: pkg/assets/bindata.go (hunk 2 above)
Comment (oglok) [reply pkg/assets/bindata.go]: Done. I did the rebase and I'm not sure why this remained as part of the PR. Sorry!

**Pull request merged**